### PR TITLE
feat: Send data

### LINKED ISSUE #4: feat: Send temperature and humidity data to server

### PR DESCRIPTION
**Pull request by joe-sacco**

### CODE REVIEW

File: src/send_data.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import requests
+import json
+import time
+
+
+URL = 'http://127.0.0.1:8000'
+
+## SEND DATA OUT
+
+valid_data = {}
```
Comment (Takashicc) [inline src/send_data.py]: Is `valid_data` needs to be global variable?

File: src/send_data.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+def create_sensor_data_dict():
+    try:
+        sensor_temp = getTemp()
+        temp_c = round(sensor_temp["temperature_c"], 1)
+        temp_f = round(sensor_temp["temperature_f"], 1)
```
Comment (Takashicc) [inline src/send_data.py:22]: Can you floor the floating number like the below implementation?

https://github.com/team-ondo/backend/blob/202e391813b8dfd0af3519a5be2366ee53f01d32/src/utils/common.py#L22-L33

File: src/send_data.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+from alarm import alarm_on
+from button import button_pressed
+import requests
+import json
+import time
+
+
+URL = 'http://127.0.0.1:8000'
+
+## SEND DATA OUT
```
Comment (Takashicc) [inline src/send_data.py]: `## SEND DATA OUT`
Remove this comment

File: src/send_data.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+            'humidity': humidity,
+            'motion': motion,
+            'alarm': alarm,
+            'button': button
+        }
+
+        valid_data = sensor_data_out
+        return sensor_data_out
+    except:
+        # pass
```
Comment (Takashicc) [inline src/send_data.py]: `# pass`
Delete this comment

File: src/send_data.py [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+        return valid_data
+
+## LOOP DATA SEND OUT
```
Comment (Takashicc) [inline src/send_data.py:42]: `## LOOP DATA SEND OUT`
Delete this comment

File: src/send_data.py [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+
+while True:
+    # create_sensor_data_dict()
+    req = requests.post(f'{URL}/sensor_data', json = create_sensor_data_dict())
+    print(req.text)
+    time.sleep(2)
```
Comment (Takashicc) [inline src/send_data.py]: Delete this script or wrap with `if __name__ == '__main__':`

File: src/server.py [hunk 7]
```diff
@@ -0,0 +1,35 @@
...
+import time
+from fastapi import FastAPI, Request
+from fastapi.middleware.cors import CORSMiddleware
+from pydantic import BaseModel
+
+
+URL = 'http://127.0.0.1:8000'
+
+## RECEIVE DATA IN
+
```
Comment (Takashicc) [inline src/server.py]: Delete this comment

File: src/server.py [hunk 8]
```diff
@@ -0,0 +1,35 @@
...
+app = FastAPI()
+
+@app.get('/')
+async def root():
+    return {'message': "Hello World"}
```
Comment (Takashicc) [inline src/server.py]: Can you delete this route?

File: src/server.py [hunk 9]
```diff
@@ -0,0 +1,35 @@
...
+class Device_Input(BaseModel):
+    temperature_c: float
+    temperature_f: float
+    humidity: float
+    motion: str
+    alarm: str
+    button: str
+
+@app.post('/sensor_data')
+async def send_data_post(device_input: Device_Input):
```
Comment (Takashicc) [inline src/server.py]: Add `response_model`.
```python
@app.post('/sensor_data/', response_model=Device_Input)
```

**Review by Takashicc — CHANGES_REQUESTED** [review]

File: src/send_data.py (hunk 2 above)
Comment (joe-sacco) [reply src/send_data.py:22]: Is there are reason for using floor instead of round?

File: src/send_data.py (hunk 6 above)
Comment (joe-sacco) [reply src/send_data.py]: We need this script to loop so it keeps sending data.    

I still don't entirely understand `if __name__ == '__main__':`.    How do I wrap the while loop inside this?

Comment (joe-sacco) [reply src/send_data.py]: like this?:

`if __name__ == '__main__':`
    while True:
        # create_sensor_data_dict()
        req = requests.post(f'{URL}/sensor_data', json = create_sensor_data_dict())
        print(req.text)
        time.sleep(2)

File: src/send_data.py (hunk 5 above)
Comment (joe-sacco) [reply src/send_data.py:42]: I'm keeping this comment so I understand what it's doing

File: src/send_data.py (hunk 3 above)
Comment (joe-sacco) [reply src/send_data.py]: I'm keeping this comment so I understand what it's doing

File: src/send_data.py (hunk 4 above)
Comment (joe-sacco) [reply src/send_data.py]: Done

File: src/server.py (hunk 8 above)
Comment (joe-sacco) [reply src/server.py]: Done

File: src/server.py (hunk 7 above)
Comment (joe-sacco) [reply src/server.py]: I'm keeping this comment so I understand what it's doing

File: src/send_data.py (hunk 2 above)
Comment (Takashicc) [reply src/send_data.py:22]: There is no particularly strong reason.
Do you want to use round?

File: src/send_data.py (hunk 6 above)
Comment (Takashicc) [reply src/send_data.py]:
```python
if __name__ == '__main__':
  while True:
    # create_sensor_data_dict()
    req = requests.post(f'{URL}/sensor_data', json = create_sensor_data_dict())
    print(req.text)
    time.sleep(2)
```

File: src/send_data.py (hunk 1 above)
Comment (joe-sacco) [reply src/send_data.py]: We removed this.

File: src/send_data.py (hunk 2 above)
Comment (joe-sacco) [reply src/send_data.py:22]: I think if they are similar we can use round, unless there's a good reason to change.

File: src/send_data.py (hunk 6 above)
Comment (joe-sacco) [reply src/send_data.py]: Done

File: src/server.py (hunk 9 above)
Comment (joe-sacco) [reply src/server.py]: Done

File: src/send_data.py [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+import time
+import datetime
+
+URL = 'http://127.0.0.1:8000'
+DEVICE_ID = '1'
+
+## SEND DATA OUT   
+def create_sensor_data_dict():
+    sensor_temp = getTemp()
+
```
Comment (Takashicc) [inline src/send_data.py]: `sensor_temp = getTemp()`
`getTemp()` will raise an error. You must try catch this too.

File: src/send_data.py [hunk 11]
```diff
@@ -0,0 +1,51 @@
...
+            'temperature_c': temp_c,
+            'temperature_f': temp_f,
+            'humidity': humidity,
+            'motion': motion,
+            'alarm': alarm,
+            'button': button,
+            'created_at': timestamp
+        }
+
+        return sensor_data_out
```
Comment (Takashicc) [inline src/send_data.py]: You can immediately return this dictionary.
```python
return {
    'temperature_c': temp_c,
    'temperature_f': temp_f,
    'humidity': humidity,
    'motion': motion,
    'alarm': alarm,
    'button': button,
    'created_at': timestamp
}
```

**Review by Takashicc — CHANGES_REQUESTED** [review]

File: src/send_data.py (hunk 11 above)
Comment (joe-sacco) [reply src/send_data.py]: Done

File: src/send_data.py (hunk 10 above)
Comment (joe-sacco) [reply src/send_data.py]: Done

**Review by Takashicc — APPROVED** [review]

**Pull request merged**